src/DependencyInjection/DI/src/ServiceLookup/ILEmit/ILEmitResolverBuilder.cs

```diff
@@ -49,25 +56,71 @@ private class ILEmitResolverBuilderRuntimeContext
             _runtimeResolver = runtimeResolver;
             _serviceScopeFactory = serviceScopeFactory;
             _rootScope = rootScope;
+            _scopeResolverCache = new ConcurrentDictionary<ServiceCacheKey, GeneratedMethod>();
         }
 
         public Func<ServiceProviderEngineScope, object> Build(ServiceCallSite callSite)
         {
-            if (callSite.Cache.Location == CallSiteResultCacheLocation.Root)
```

Comment:
Might be worth an assert that `callSite.Cache.Location != CallSiteResultCacheLocation.Root` if only to make it obvious what's going on in the future.

Reply:
Good point.